editor/editor_help.cpp

```diff
@@ -3334,6 +3370,166 @@ EditorHelpTooltip::EditorHelpTooltip(const String &p_text, const String &p_custo
 	get_rich_text()->set_custom_minimum_size(Size2(360 * EDSCALE, 0));
 }
 
+#if defined(MODULE_GDSCRIPT_ENABLED) || defined(MODULE_MONO_ENABLED)
+/// EditorHelpHighlighter ///
+
+EditorHelpHighlighter *EditorHelpHighlighter::singleton = nullptr;
+
+void EditorHelpHighlighter::create_singleton() {
+	ERR_FAIL_COND(singleton != nullptr);
+	singleton = memnew(EditorHelpHighlighter);
+}
+
+void EditorHelpHighlighter::free_singleton() {
+	ERR_FAIL_NULL(singleton);
+	memdelete(singleton);
+	singleton = nullptr;
+}
+
+EditorHelpHighlighter *EditorHelpHighlighter::get_singleton() {
+	return singleton;
+}
+
+EditorHelpHighlighter::HighlightData EditorHelpHighlighter::_get_highlight_data(Language p_language, const String &p_source, bool p_use_cache) {
```

Comment:
This method probably needs a mutex, since a single `Script` + `TextEdit` + `EditorSyntaxHighlighter` are used (within one language). And probably different `EditorHelp`s work in different threads? But I didn't notice any problems during testing however.